python/ray/dag/dag_node.py

```diff
@@ -63,11 +63,13 @@ def __init__(
 
         # The list of nodes that use this DAG node as an argument.
         self._downstream_nodes: List["DAGNode"] = []
-        # The list of nodes that this DAG node uses as an argument.
-        self._upstream_nodes: List["DAGNode"] = self._collect_upstream_nodes()
 
         # UUID that is not changed over copies of this node.
         self._stable_uuid = uuid.uuid4().hex
+
+        # The list of nodes that this DAG node uses as an argument.
+        self._upstream_nodes: List["DAGNode"] = self._collect_upstream_nodes()
```

Comment:
I've decided to move this function below `_stable_uuid` so that it won't throw an error when `print(self)` is called in `_collect_upstream_nodes` for debugging, as `self._upstream_nodes` doesn't exist yet.